Dispersing artillery shells

9Valjew opened the pull request:
Allows crafting of dispersible canisters with express shells
Does not return the empty canister
Does not spawn lander shell if not needed
Copies the chlorine mist spawning because I cba writing that part out

9Valjew committed:
Allowed crafting cargo shells with dispersible canisters (not glands fuck you)

9Valjew committed:
Mist spawning and it fucking works

9Valjew committed:
The great butchering

9Valjew committed:
Merge branch 'HbmMods:master' into spraying-shells

HbmMods commented on src/main/java/com/hbm/crafting/handlers/CargoShellCraftingHandler.java:
```diff
@@ -56,6 +56,9 @@ public ItemStack getCraftingResult(InventoryCrafting inventory) {
 				cargo = copy;
 			}
+			if(stack.getItem() == ModItems.disperser_canister) {
+				stack.getItem().setContainerItem(null); //removes the empty canister from the output
```
doing it like this will make it so that the disperser canister won't have a container item at all anymore after this recipe is used once

9Valjew commented:
Thats whar I thought but apprently it still keeps it (somehow)
You are free to check by all means that was implemented just as a check and
ended up working

On Fri, Feb 2, 2024, 09:06 HbmMods ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/main/java/com/hbm/crafting/handlers/CargoShellCraftingHandler.java
> <https://github.com/HbmMods/Hbm-s-Nuclear-Tech-GIT/pull/1320#discussion_r1475640808>
> :
>
> > @@ -56,6 +56,9 @@ public ItemStack getCraftingResult(InventoryCrafting inventory) {
>  				copy.stackSize = 1;
>  				cargo = copy;
>  			}
> +			if(stack.getItem() == ModItems.disperser_canister) {
> +				stack.getItem().setContainerItem(null); //removes the empty canister from the output
>
> doing it like this will make it so that the disperser canister won't have
> a container item at all anymore after this recipe is used once
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/HbmMods/Hbm-s-Nuclear-Tech-GIT/pull/1320#pullrequestreview-1858513089>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APKLT5FANXCGA3L7KSRJICLYRSGAHAVCNFSM6AAAAABCTR474GVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTQNJYGUYTGMBYHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

9Valjew commented:
unless you mean container item as in in future crafts which makes sense
for now, there are no recipes that need a disperser canister, and i cant think of a recipe which would need one
it still works as a fluid container, but idk, its your choice

9Valjew commented:
(Also I have to say that in my opinion it should not have a container item by the literal dispersing nature, if I use it, for example, in a chemical warhead or whatnot, I seek the disperser quality, otherwise I could just have used 2 universal fluid tanks if I wanted 2000mb)

9Valjew commented:
aoogh I will try to find a workaround

9Valjew committed:
Ok fuck container items its a literal disperser but I cleaned it up a bit for now so the removal of the container item isnt held inside the cargo crafter

9Valjew committed:
Merge branch 'Arty-Spray' into spraying-shells

9Valjew committed:
cleaning up and why the fuck was I on two branches

9Valjew commented:
Ok now I:
Changed the disperser canister to have no container item because fuck it?
cleaned up the code handling the disperser canister being a container item in the cargo handler
Bobbert I am begging you I am on my knees

Vaern commented:
![image](https://github.com/HbmMods/Hbm-s-Nuclear-Tech-GIT/assets/90118520/2b684313-1b50-40ba-94db-33ec2e00dce3)

HbmMods commented:
![image](https://github.com/HbmMods/Hbm-s-Nuclear-Tech-GIT/assets/16674258/18dfa27f-cb06-4416-948a-7afefac0122d)

the true question is why we even need an artillery shell like that at all, and why it has to be dispersers inside cargo shells, which isn't even documented, instead of having disperser shells
